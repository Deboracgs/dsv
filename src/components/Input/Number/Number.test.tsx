import { fireEvent, render } from "@testing-library/react";
import WrapperMock from "../../../utils/wrapperMock";
import InputNumber, { InputNumberProps } from "./Number";

describe("Input Number Component", () => {
  const defaultProps: InputNumberProps = {
    id: "unit-number", 
    defaultValue: 0,
    onChange: jest.fn()
  };

  const rendered = (props: InputNumberProps) =>
    render(
      <WrapperMock>
        <InputNumber {...props} />
      </WrapperMock>
    );

  it("render", () => {
    const onChangeSpy = jest.spyOn(defaultProps, "onChange");
    const { asFragment, container } = rendered(defaultProps);

    const element = container.querySelector("#unit-number") as HTMLInputElement;

    expect(element).toBeDefined();
    expect(element.defaultValue).toBe("0");
    fireEvent.change(element, {target: {value: 3}});
    
    expect(onChangeSpy).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });

});
