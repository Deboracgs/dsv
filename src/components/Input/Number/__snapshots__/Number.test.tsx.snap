// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Number Component render 1`] = `
<DocumentFragment>
  <div
    data-testid="input-number-unit-number"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="unit-number"
          name="input-number-unit-number"
          type="number"
          value="0"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-hdw1oc"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
